### ISSUE TITLE
min-st-cut interface

### ISSUE THREAD

**Issue by DerThorsten** [DerThorsten/nifty#56]
implement min-st-cut interface similar to multicut/lifted multucut

**Comment by DerThorsten** [comment]
@ahmadnish  you can look into this.
The goal is to implement minstcut (also known as GraphCut ) functionality similar to multicut, mincut, lifted_multicut.
In fact,  mincut and minstcut  are very simmilar.
you just need to add unaries to mincut.
This just implements a Potts Graphical model with binary variables.
Do you think you can handle this?

One Needs to implent:
- MinStCutObjective
- MinStCutFactoryBase
- MinStCutFactory
- MinStCutBase
- Visitors
- If you are very very brave you can also try to implement a wrapper to 
 a kolmogorov maxflow implementation but consider this task to be very optional

For most of these things one should derive from the matching base classes.
Basically look into the  mincut and do the sane thing for a minstcut.

For mincut, mulicut, and lifted_multicut one just needs weights which are the ``beta`` of the potts regularizer.
For a minStCut we need a ``NodeMap`` which stores a `std::pair<float,float>` (or simmilar).
This can be realized by adding an other template to MinStCutObjective in addition to the `WEIGHT_TYPE` (the edge weights).
The name of the template could be `UNARIES`
